### PR TITLE
Getting started tests

### LINKED ISSUE #734: Steps in Getting Started documentation may break without getting noticed
## Problem

I've performed the current [getting started steps](https://agoric.com/documentation/getting-started/)
I've come across an error at `agoric deploy`

[I'm not the only one](https://github.com/Agoric/agoric-sdk/issues/715) and it's not the first time. Given agoric-cli depends on pretty much every (?!) package of agoric-sdk, i'm inclined to think that it will happen again. But since the steps described in the doc are in the doc, there is no way to know they have broken

This will result in a frustrating developer experience i imagine

## Potential solutions

Write automated tests for what's in the doc
This way, if the tests pass, it should give strong assurance that the doc document something that works as expected
And fix things so the tests pass otherwise

### PR DESCRIPTION
**Pull request by michaelfig**
This runs the "Getting started" documentation tests as part of the `agoric-cli` tests.

Closes #734, Closes #760

### CODE REVIEW

**Comment by michaelfig** [comment]
It may be desirable to take this entire test and put it back directly into the CI, as #760 has done.  I just started working with what I had for `test-workflow.js`.

Or at least, don't run this test via the normal unit tests, just from CI.

File: packages/agoric-cli/test/test-workflow.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+import { producePromise } from '@agoric/produce-promise';
+import { request } from 'http';
 
 import { spawn } from 'child_process';
 
-const SIMPLEST_TEMPLATE = 'dapp-encouragement';
-
 test('workflow', async t => {
   try {
+    // FIXME: Do a search for an unused port.
```
Comment (Chris-Hibbert) [inline packages/agoric-cli/test/test-workflow.js]: or allow specification with a parameter.

**Review by Chris-Hibbert — COMMENTED** [review]
I'd be happy to have it in CI as long as it's not flakey. When this kind of test is flakey (which isn't uncommon), it becomes a source of noise. How many times has it been run successfully?

**Comment by michaelfig** [comment]
> I'd be happy to have it in CI as long as it's not flakey. When this kind of test is flakey (which isn't uncommon), it becomes a source of noise. How many times has it been run successfully?

It's not flakey unless the user runs it manually (and has `agoric` instances currently running).  So, putting it in CI is reasonable.

I've run it tens of times without flakes.

**Comment by michaelfig** [comment]
> Or at least, don't run this test via the normal unit tests, just from CI.

Done, for efficiency reasons, not flake reasons.

**Review by katelynsills — APPROVED** [review]
This looks great to me! Can we write up what exactly this tests somewhere? I see it in the comments, but it looks like there are no tests of whether the deploy works successfully (calling a publicAPI method on a contract, etc). It seems to stop at whether deploy.js errors or not, is that right? Having a clear picture of what is not tested (or is a future TODO to test) would be great.

**Comment by DavidBruant** [comment]
As i said in Keybase, i think that this version of the tests is better than what i had written, because this version works and mine didin't

I haven't checked, but i believe these tests will capture the same sort of problems i've encountered

Thanks for working on this!

**Pull request merged**